arches/app/media/js/viewmodels/workflow.js

```diff
@@ -37,9 +37,16 @@ define([
         });
 
         this.ready = ko.observable(false);
-        this.ready.subscribe(function() {
-            var components = _.unique(self.steps.map(function(step) {return step.component;}));
-            require(components, function() { self.initialize(); });
+
+        this.nameReady = ko.computed(function(){
+            return self.ready() && self.workflowName();
+        })
+
+        this.nameReady.subscribe(function(ready) {
+            if (ready){
+                var components = _.unique(self.steps.map(function(step) {return step.component;}));
+                require(components, function() { self.initialize(); });    
+            }
```

Comment:
Love the code overall, this is my last and only nit, and it's more about the architecture present around the code written here.

`ready` is a remnant of v1 workflows, and will likely be factored out in the near future. Instead the component will just function off `loading`.  It's only ever called in the `${project}-workflow.js` file and it's easy to change the code involving it to use different logic. It's counter-intuitive to have both `loading` and `ready` logic in the same component and is potentially asking for trouble down the line.

Just talking about future direction, not asking for those changes in this ticket :) 

Following this logic, would it be possible to drop the `ready` nomenclature ? Sommit like:

```
        this.ready = ko.observable(false);
        
        this.workflowName = ko.observable();
        this.workflowName.subscribe(function(workflowName) {
            if (workflowName && self.ready()) {
                var components = _.unique(self.steps.map(function(step) {return step.component;}));
                require(components, function() { self.initialize(); });
            }
        });

```

Line 1 on `workflow.htm` will need to be changed to accommodate the new code introduced here, be it yours or mine. Something like `ready() && workflowName()`

As always if you disagree or feel it's better to just push this out the door, I'm always open to discussion and opposing opinions 👍

Reply:
good suggestion. fixed accordingly.